packages/block-editor/src/hooks/anchor.js

```diff
@@ -64,71 +127,19 @@ export const withInspectorControl = createHigherOrderComponent(
 	( BlockEdit ) => {
 		return ( props ) => {
 			const hasAnchor = hasBlockSupport( props.name, 'anchor' );
-			const blockEditingMode = useBlockEditingMode();
-
-			if ( hasAnchor && props.isSelected ) {
-				const isWeb = Platform.OS === 'web';
-				const textControl = (
-					<TextControl
-						__nextHasNoMarginBottom
-						className="html-anchor-control"
-						label={ __( 'HTML anchor' ) }
-						help={
-							<>
-								{ __(
-									'Enter a word or two — without spaces — to make a unique web address just for this block, called an “anchor.” Then, you’ll be able to link directly to this section of your page.'
-								) }
-
-								{ isWeb && (
-									<ExternalLink
-										href={ __(
-											'https://wordpress.org/documentation/article/page-jumps/'
-										) }
-									>
-										{ __( 'Learn more about anchors' ) }
-									</ExternalLink>
-								) }
-							</>
-						}
-						value={ props.attributes.anchor || '' }
-						placeholder={ ! isWeb ? __( 'Add an anchor' ) : null }
-						onChange={ ( nextValue ) => {
-							nextValue = nextValue.replace( ANCHOR_REGEX, '-' );
-							props.setAttributes( {
-								anchor: nextValue,
-							} );
-						} }
-						autoCapitalize="none"
-						autoComplete="off"
-					/>
-				);
-
-				return (
-					<>
-						<BlockEdit { ...props } />
-						{ isWeb && blockEditingMode === 'default' && (
-							<InspectorControls group="advanced">
-								{ textControl }
-							</InspectorControls>
-						) }
-						{ /*
-						 * We plan to remove scoping anchors to 'core/heading' to support
-						 * anchors for all eligble blocks. Additionally we plan to explore
-						 * leveraging InspectorAdvancedControls instead of a custom
-						 * PanelBody title. https://github.com/WordPress/gutenberg/issues/28363
-						 */ }
-						{ ! isWeb && props.name === 'core/heading' && (
-							<InspectorControls>
-								<PanelBody title={ __( 'Heading settings' ) }>
-									{ textControl }
-								</PanelBody>
-							</InspectorControls>
-						) }
-					</>
-				);
-			}
 
-			return <BlockEdit { ...props } />;
+			return (
+				<>
+					<BlockEdit { ...props } />
+					{ hasAnchor && props.isSelected && (
```

Comment:
This could be further micro-optimized to evaluate `hasAnchor` only after `isSelected` turns out to be `true`.

Reply:
Updated in [fdf2f10](https://github.com/WordPress/gutenberg/pull/55721/commits/fdf2f1027a6e74e9fc24168ade61dacd67420a90).